### PR TITLE
feat: add some tables for `information_schema`

### PR DESCRIPTION
**Pull request by dimbtp**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add some tables that is `not implemented` for `information_schema`, includes: `GLOBAL_STATUS, OPTIMIZER_TRACE, PARAMETERS, PROFILING, REFERENTIAL_CONSTRAINTS, ROUTINES, SCHEMA_PRIVILEGES, SESSION_STATUS, TABLE_PRIVILEGES, TRIGGERS`.

As for `TABLESPACES` table, according to [mysql doc](https://dev.mysql.com/doc/refman/8.2/en/information-schema-tablespaces-table.html): `This table is unused. It is deprecated; expect it to be removed in a future MySQL release. Other INFORMATION_SCHEMA tables may provide related information`. Should I skip this table? 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)

#2931

### CODE REVIEW

File: src/catalog/src/information_schema/memory_table/tables.rs [hunk 1]
```diff
@@ -227,6 +227,187 @@ pub fn get_schema_columns(table_name: &str) -> (SchemaRef, Vec<VectorRef>) {
             vec![],
         ),
 
+        OPTIMIZER_TRACE => (
```
Comment (killme2008) [inline src/catalog/src/information_schema/memory_table/tables.rs:230]: Can we add a TODO here or create an issue for `OPTIMIZER_TRACE`? GreptimeDB already has query tracing, we can support it later cc @Taylor-lagrange

File: src/catalog/src/information_schema/memory_table/tables.rs [hunk 2]
```diff
@@ -227,6 +227,187 @@ pub fn get_schema_columns(table_name: &str) -> (SchemaRef, Vec<VectorRef>) {
...
+                bigint_column("PAGE_FAULTS_MINOR"),
+                bigint_column("SWAPS"),
+                string_column("SOURCE_FUNCTION"),
+                string_column("SOURCE_FILE"),
+                bigint_column("SOURCE_LINE"),
+            ],
+            vec![],
+        ),
+
+        REFERENTIAL_CONSTRAINTS => (
```
Comment (killme2008) [inline src/catalog/src/information_schema/memory_table/tables.rs:290]: Can we add a TODO? If we support the foreign key constraint in the future, we must implement this table.

File: src/catalog/src/information_schema/memory_table/tables.rs [hunk 3]
```diff
@@ -227,6 +227,187 @@ pub fn get_schema_columns(table_name: &str) -> (SchemaRef, Vec<VectorRef>) {
...
+                string_column("SQL_MODE"),
+                string_column("DEFINER"),
+                string_column("CHARACTER_SET_CLIENT"),
+                string_column("COLLATION_CONNECTION"),
+                string_column("DATABASE_COLLATION"),
+            ],
+            vec![],
+        ),
+
+        GLOBAL_STATUS => (
```
Comment (killme2008) [inline src/catalog/src/information_schema/memory_table/tables.rs:395]: We want to store the internal metrics for `global_stauts` and `session_status`, but it's not decided. Could you leave a TODO?

**Review by killme2008 — APPROVED** [review]
Happy new year, a great work!  Thank you.

**Comment by evenyag** [comment]
> As for `TABLESPACES` table, according to [mysql doc](https://dev.mysql.com/doc/refman/8.2/en/information-schema-tablespaces-table.html): `This table is unused. It is deprecated; expect it to be removed in a future MySQL release. Other INFORMATION_SCHEMA tables may provide related information`. Should I skip this table?


I think we could ignore this table. cc @sunng87 @killme2008

**Comment by killme2008** [comment]
> > As for `TABLESPACES` table, according to [mysql doc](https://dev.mysql.com/doc/refman/8.2/en/information-schema-tablespaces-table.html): `This table is unused. It is deprecated; expect it to be removed in a future MySQL release. Other INFORMATION_SCHEMA tables may provide related information`. Should I skip this table?
> 
> I think we could ignore this table. cc @sunng87 @killme2008

Yes， of course

File: src/catalog/src/information_schema/memory_table/tables.rs [hunk 4]
```diff
@@ -227,6 +227,191 @@ pub fn get_schema_columns(table_name: &str) -> (SchemaRef, Vec<VectorRef>) {
             vec![],
         ),
 
+        // TODO: GreptimeDB already has query tracing, support later. cc @Taylor-lagrange
```
Comment (evenyag) [inline src/catalog/src/information_schema/memory_table/tables.rs:230]: I posted an issue to track this, we can remove this TODO from the comment. https://github.com/GreptimeTeam/greptimedb/issues/3064
```suggestion
```

**Review by evenyag — APPROVED** [review]
Mostly LGTM.

**Comment by killme2008** [comment]
I commit the suggestion and it will be merged automatically.

**Pull request merged**